pkg/utils/gardener/shoot.go

```diff
@@ -334,6 +334,7 @@ func (s *ShootAccessSecret) Reconcile(ctx context.Context, c client.Client) erro
 	_, err := controllerutils.GetAndCreateOrMergePatch(ctx, c, s.Secret, func() error {
 		s.Secret.Type = corev1.SecretTypeOpaque
 		metav1.SetMetaDataLabel(&s.Secret.ObjectMeta, resourcesv1alpha1.ResourceManagerPurpose, resourcesv1alpha1.LabelPurposeTokenRequest)
+		metav1.SetMetaDataLabel(&s.Secret.ObjectMeta, resourcesv1alpha1.ResourceManagerClass, resourcesv1alpha1.ResourceManagerClassShoot)
```

Comment:
Would it make sense to introduce an `ResourceManagerClass` option to the [RenewAccessSecrets](https://github.com/gardener/gardener/blob/27f0329a86b1a63c052cadbe45ee75a1e20f1e15/pkg/utils/gardener/tokenrequest/secrets.go#L60) helper function too? We use it when the service account signing keys are rotated. Different source cluster probably rotate their credentials independent from each other.

When we enable `tokenrequestor.Reconciler.Class` option in `pkg/resourcemanager/controller/add.go` we could enabled the option where `RenewAccessSecrets` is used too.

Reply:
Good suggestion, there should be a mechanism for renewing all garden access secrets automatically. I added a corresponding task to https://github.com/gardener/gardener/issues/8001.

However, I'm hesitant to piggyback on the `renew-kubeconfig` operation because
- it only works for `ManagedSeeds` but not for non-managed `Seeds`
- the gardenlet's kubeconfig credentials are signed by a different authority than the garden access tokens (ca-client vs. ServiceAccount signing key), hence operators might want/need to renew the credentials separately.

Instead, I propose renewing all garden access tokens when the `Seed` is annotated with `gardener.cloud/operation=renew-garden-access-tokens`.
If you agree, I can add a corresponding implementation in a dedicated PR.

Reply:
@rfranzke @oliver-goetz if you don't object, I suggest moving on with this PR and adding the rotation mechanism in a dedicated PR as proposed :)

Reply:
fine for me